doc/architectural/compilation-and-development.md

```diff
@@ -270,3 +270,28 @@ branch, e.g. to run it on lines that have been changed from `develop`:
 
 There are also instructions for adding this as a git pre-commit hook in
 [CODING_STANDARD.md](https://github.com/diffblue/cbmc/blob/develop/CODING_STANDARD.md#pre-commit-hook-to-run-cpplint-locally).
+
+
+\section compilation-and-development-section-time-profiling Time profiling
+
+To do time profiling with a tool like `gprof`, the flags `-g` and `-pg` must be
+used which compiling, and `-pg` must be used when linking. If you are building
+with cmake you can just add "-Denable_profiling=1" to your cmake invocation, and
+reload cmake before building your desired binary.
+
+Run your binary as normal. A file called gmon.out will be created in the
+working directory of the binary at the end of execution. In most instances this
+will be the same as the working directory at the beginning of execution. It is
+also possible to choose the output location using an environment variable -
+search online for more details.
```

Comment:
:bulb: A link would be more welcoming than telling people to just "google it". :slightly_smiling_face:

Reply:
I've google it myself and put the answer in